packages/ngtools/webpack/src/gather_diagnostics.ts

```diff
@@ -52,34 +58,44 @@ export function gatherDiagnostics(
     }
   }
 
+  const gatherSyntacticDiagnostics = mode == DiagnosticMode.All || mode == DiagnosticMode.Syntactic;
```

Comment:
With the above you can just use `mode & DiagnosticMode.Syntactic != 0`

Reply:
Updated to `(mode & DiagnosticMode.Syntactic) != 0` because `!=` seems to have precedence over `&`.

Reply:
JavaScript!